yt/fields/local_fields.py

```diff
@@ -29,10 +27,6 @@ def add_field(self, name, function=None, sampling_type=None, **kwargs):
             else:
                 sampling_type = "particle"
         if sampling_type is None:
```

Comment:
We should instead set `sampling_type = "cell"` in the function definition.

Reply:
Indeed, I always forget which builtin iterable types are mutable or not, but str isn't so it's okay to use default str values.

Reply:
Actually I think the parameter should have no default value at all in order to embrace the vision that was described at the time of yt 3.4
Right now I'm testing against using "cell" as a default, since the reimplementation should slightly affect the function's behaviour. If all goes well I'll also try to remove the default value all together.